### PR TITLE
BIG 10 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/loot_the_key_to_everything.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+ManaCost:G U R
+Types:Legendary Creature Beast Noble
+PT:1/2
+K:Ward:1
+T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigDig | TriggerDescription$ At the beginning of your upkeep, exile the top X cards of your library, where X is the number of card types among other nonland permanents you control. You may play those cards this turn.
+SVar:TrigDig:DB$ Dig | Defined$ You | DigNum$ X | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBEffect
+SVar:DBEffect:DB$ Effect | RememberObjects$ RememberedCard | StaticAbilities$ Play | SubAbility$ DBCleanup | ForgetOnMoved$ Exile
+SVar:Play:Mode$ Continuous | MayPlay$ True | EffectZone$ Command | Affected$ Card.IsRemembered | AffectedZone$ Exile | Description$ You may play those cards this turn.
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:X:Count$CardTypes.Permanent.nonLand+Other+Youctrl
```
Comment (Simisays) [inline forge-gui/res/cardsfolder/upcoming/loot_the_key_to_everything.txt:11]: This is not working. Thoughts ?

File: forge-gui/res/cardsfolder/upcoming/greeds_gambit.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+SVar:DBGainLife:DB$ GainLife | Defined$ You | LifeAmount$ 6 | SubAbility$ DBToken
+SVar:DBToken:DB$ Token | TokenAmount$ 3 | TokenScript$ b_2_1_bat_flying | TokenOwner$ You
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigDiscard | TriggerDescription$ At the beginning of your end step, you discard a card, lose 2 life, and sacrifice a creature.
+SVar:TrigDiscard:DB$ Discard | NumCards$ 1 | Mode$ TgtChoose | SubAbility$ DBLoseLife
+SVar:DBLoseLife:DB$ LoseLife | Defined$ You | LifeAmount$ 2 | SubAbility$ DBSacrifice
+SVar:DBSacrifice:DB$ Sacrifice | SacValid$ Creature
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Any | ValidCard$ Card.Self | Execute$ TrigDiscardThree | TriggerDescription$ When CARDNAME leaves the battlefield, you discard three cards, lose 6 life, and sacrifice three creatures.
+SVar:TrigDiscardThree:DB$ Discard | NumCards$ 3 | Mode$ TgtChoose | SubAbility$ DBLoseLifeThree
+SVar:DBLoseLifeThree:DB$ LoseLife | Defined$ You | LifeAmount$ 6 | SubAbility$ DBSacrificeThree
+SVar:DBSacrificeThree:DB$ Sacrifice | SacValid$ Creature | SacMessage$ Sacrifice three creatures | Amount$ 3
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/greeds_gambit.txt:15]: I don't think this SacMessage is right

File: forge-gui/res/cardsfolder/upcoming/generous_plunderer.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+Types:Creature Human Rogue
+PT:2/2
+K:Menace
+T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigTreasure | OptionalDecider$ You  | TriggerDescription$ At the beginning of your upkeep, you may create a Treasure token. When you do, target opponent creates a tapped Treasure token.
+SVar:TrigTreasure:DB$ Token | TokenScript$ c_a_treasure_sac | RememberTokens$ True | SubAbility$ DBImmediateTrigger
+SVar:DBImmediateTrigger:DB$ ImmediateTrigger | ConditionDefined$ Remembered | ConditionPresent$ Card | Execute$ TrigToken | RememberObjects$ RememberedCard | SubAbility$ DBCleanup | SpellDescription$ When you do, target opponent creates a tapped Treasure token.
+SVar:TrigToken:DB$ Token | TokenAmount$ 1 | ValidTgts$ Opponent | TokenOwner$ Targeted | TokenScript$ c_a_treasure_sac | TokenTapped$ True 
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigDamage | TriggerDescription$ Whenever CARDNAME attacks, it deals damage to defending player equal to the number of artifacts they control.
+SVar:TrigDamage:DB$ DealDamage | Defined$ TriggeredDefender | NumDmg$ X
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/generous_plunderer.txt:12]: wrong defined

File: forge-gui/res/cardsfolder/upcoming/greeds_gambit.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+DeckHas:Ability$LifeGain|Tokens|Sacrifice & Type$Bat
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/greeds_gambit.txt:16]: bad ability

File: forge-gui/res/cardsfolder/upcoming/hostile_investigator.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+Name:Hostile Investigator
+ManaCost:3 B
+Types:Creature Ogre Rogue Detective
+PT:4/3
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDiscard | TriggerDescription$ When CARDNAME enters the battlefield, target opponent discards a card.
+SVar:TrigDiscard:DB$ Discard | ValidTgts$ Opponent | NumCards$ 1 | Mode$ TgtChoose
+T:Mode$ DiscardedAll | ValidPlayer$ Player | ActivationLimit$ 1 | FirstTime$ True | ValidCard$ Card | TriggerZones$ Battlefield | Execute$ TrigInvestigate | TriggerDescription$ Whenever one or more players discard one or more cards, investigate. This ability triggers only once each turn. (Create a Clue token. It's an artifact with "{2}, Sacrifice this artifact: Draw a card.")
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/hostile_investigator.txt:7]: no firsttime

File: forge-gui/res/cardsfolder/upcoming/lost_jitte.txt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+Name:Lost Jitte
+ManaCost:1
+Types:Legendary Artifact Equipment
+K:Equip:1
+T:Mode$ DamageDealtOnce | CombatDamage$ True | ValidSource$ Creature.EquippedBy | Execute$ TrigPutCounter | TriggerZones$ Battlefield | TriggerDescription$ Whenever equipped creature deals combat damage, put a charge counter on CARDNAME.
+SVar:TrigPutCounter:DB$ PutCounter | CounterType$ CHARGE | CounterNum$ 1
+A:AB$ Charm | Cost$ SubCounter<1/CHARGE> | Choices$ DBUntap,DBCantBlock,DBPutCounter | Defined$ You
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/lost_jitte.txt:7]: defined?

File: forge-gui/res/cardsfolder/upcoming/memory_vessel.txt [hunk 7]
```diff
@@ -0,0 +1,10 @@
+Name:Memory Vessel
+ManaCost:3 R R
+Types:Artifact
+A:AB$ Dig | Cost$ T Exile<1/CARDNAME> | Defined$ Player | DigNum$ 7 | ChangeNum$ All | DestinationZone$ Exile | SorcerySpeed$ True | RememberChanged$ True | SubAbility$ DBEffect | SpellDescription$ Each player exiles the top seven cards of their library. Until your next turn, players may play cards they exiled from their library this way, and they can't play cards from their hand. Activate only as a sorcery.
+SVar:DBEffect:DB$ Effect | StaticAbilities$ STPlay,NoLand,NoCast | RememberObjects$ Remembered | Duration$ UntilYourNextTurn | SubAbility$ DBCleanup | ForgetOnMoved$ Exile
+SVar:STPlay:Mode$ Continuous | MayPlay$ True | MayPlayPlayer$ CardOwner | EffectZone$ Command | Affected$ Card.IsRemembered | AffectedZone$ Exile | Description$ Until your next end step, each player may play the card they exiled this way.
+SVar:NoLand:Mode$ CantPlayLand | Player$ Player | Origin$ Hand | Description$ Players can't play lands from their hand.
+SVar:NoCast:Mode$ CantBeCast | ValidCard$ Card | Caster$ You | Origin$ Hand | Description$ Players can't cast spells from their hand.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/memory_vessel.txt:8]: Valid incomplete

**Review by tool4ever — APPROVED** [review]

**Pull request merged**